src/tir/transforms/lower_sparse_tir.cc

```diff
@@ -83,178 +83,217 @@ Map<Var, Buffer> UpdateBufferMap(PrimFunc f) {
   return std::move(updater.buffer_map_);
 }
 
+/* Storing the context information of a sparse block.
+ */
+class SparseBlockCtx {
+ public:
+  explicit SparseBlockCtx(String blk_name, Array<SpIterVar> sp_iter_vars, const AxisTree& tree)
+      : blk_name_(std::move(blk_name)) {
+    // initialize sparse iter var dependency map.
+    std::unordered_map<String, SpIterVar> axis_name_sp_iter_map_;
+    for (const SpIterVar& sp_iter_var : sp_iter_vars) {
+      axis_name_sp_iter_map_[sp_iter_var->axis->name] = sp_iter_var;
+    }
+
+    for (const SpIterVar& sp_iter_var : sp_iter_vars) {
+      String axis_name = sp_iter_var->axis->name;
+      const SpIterVarNode* node = sp_iter_var.get();
+      if (axis_name.size() > 6 && std::string(axis_name).compare(axis_name.size() - 6, 6, "_dense") == 0) {
+        // ends with "_dense", the axis is generated via to_dense
+        parent_sp_iter_var_[node] = sp_iter_var;
+      } else {
+        auto opt = tree->parent.Get(axis_name);
+        CHECK(opt.defined()) << "Cannot find parent of axis " << axis_name << ".";
+        String parent_axis_name = opt.value();
+        if (parent_axis_name != "root") {
+          auto it = axis_name_sp_iter_map_.find(parent_axis_name);
+          CHECK(it != axis_name_sp_iter_map_.end())
+              << "Cannot find sparse iter vars corresponding to parent axis " << parent_axis_name
+              << " in current sparse block " << blk_name_;
+          parent_sp_iter_var_[node] = it->second;
+        }
+      }
+    }
+  }
+
+  void add_sp_iter(const VarNode* node, const SpIterVar& sp_iter_var) {
```

Comment:
CamelCase for methods

```suggestion
  void AddSpIter(const VarNode* node, const SpIterVar& sp_iter_var) {
```